### PR TITLE
perf(insights): Speed up loading properties

### PR DESCRIPTION
**Pull request by timgl**
## Problem

Loading properties  is _painful_, takes about 7 seconds on our team. Reason is that we didn't set an offset/limit on the query, so it was always going through 100Ks of properties for each query.

In general the way we do properties needs work. We now spend ~5 minutes loading all properties for our team into browser memory, which you really start to notice. We should probably send property definitions along with properties wherever applicable instead

## Changes

set limit/offset
also bump up the breakpoint for loading properties (and other stuff), as you very often end up loading lots in parallel.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/api/property_definition.py [hunk 1]
```diff
@@ -115,6 +115,9 @@ def get_queryset(self):
             **search_kwargs,
         }
 
+        limit = self.paginator.get_limit(self.request)
```
Comment (EDsCODE) [inline posthog/api/property_definition.py:118]: Are these params being passed already?

Comment (timgl) [reply posthog/api/property_definition.py:118]: Yep!

File: posthog/api/property_definition.py [hunk 2]
```diff
@@ -130,6 +133,7 @@ def get_queryset(self):
                             WHERE team_id = %(team_id)s AND name NOT IN %(excluded_properties)s
                              {name_filter} {numerical_filter} {search_query} {event_property_filter}
                             ORDER BY is_event_property DESC, query_usage_30_day DESC NULLS LAST, name ASC
+                            LIMIT {limit} OFFSET {offset}
```
Comment (pauldambra) [inline posthog/api/property_definition.py:136]: flyby....

I think by setting limit and offset manually here we only ever get one page.

DRF `LimitOffsetPagination` uses the queryset to count the results here https://github.com/encode/django-rest-framework/blob/master/rest_framework/pagination.py#L387

And then uses that count to decide whether there are more pages. 

It slices the queryset before loading the results. 

So if we add limit and offset to the queryset the count is always equal to one page. 

The TTFB on cloud for this endpoint is _wild_ though even so

File: cypress/e2e/a11y.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ describe('a11y', () => {
     it('home should have no accessibility violations', () => {
         cy.get('[data-attr="menu-item-projecthomepage"]').click()
         cy.injectAxe()
-        reportA11y({ includedImpacts: ['critical'] }, 'home-page-critical', false)
```
Comment (pauldambra) [inline cypress/e2e/a11y.js:7]: This silences an accessibility failure that I thought was resolved. 

I'll follow this up in separate work. 

The change to false here still runs the accessibility test but doesn't fail the test run when violations are found

**Comment by pauldambra** [comment]
@neilkakkar I think this covers the conversation here https://posthog.slack.com/archives/C0368RPHLQH/p1660727404221419

I'd be interested in releasing this so we can measure it *without trying to cache the count*

**Review by neilkakkar — APPROVED** [review]
I'm a bit confused over the Pagination class, but the querying code looks right to me!

Good call on not caching right now

File: posthog/api/property_definition.py [hunk 4]
```diff
@@ -48,6 +92,86 @@ def update(self, property_definition: PropertyDefinition, validated_data):
...
+                    "format": "uri",
+                    "example": "http://api.example.org/accounts/?{offset_param}=200&{limit_param}=100".format(
+                        offset_param=self.offset_query_param, limit_param=self.limit_query_param
+                    ),
+                },
+                "results": schema,
+            },
+        }
+
+    def get_next_link(self) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/api/property_definition.py:157]: Ah yes! I'm not removing count any more so can override way less of this.

The default implementation asks the queryset for the count and slices the results. But that's being done externally.

**Review by neilkakkar — APPROVED** [review]
Much cleaner, nice work!

**Pull request merged**